internal/app/singularity/instance_linux.go

```diff
@@ -122,19 +130,127 @@ func WriteInstancePidFile(name, pidFile string) error {
 	return nil
 }
 
+// instanceListOrError is a private function to retrieve named instances or fail if there are no instances
+// We wrap the error from instance.List to provide a more specific error message
+func instanceListOrError(instanceUser, name string) ([]*instance.File, error) {
+	ii, err := instance.List(instanceUser, name, instance.SingSubDir)
+	if err != nil {
+		return ii, fmt.Errorf("could not retrieve instance list: %w", err)
+	}
+	if len(ii) == 0 {
+		return ii, fmt.Errorf("no instance found")
+	}
+	return ii, err
+}
+
+// calculate BlockIO counts up read/write totals
+func calculateBlockIO(stats *libcgroups.BlkioStats) (float64, float64) {
+	var read, write float64
+	for _, entry := range stats.IoServiceBytesRecursive {
+		switch strings.ToLower(entry.Op) {
+		case "read":
+			read += float64(entry.Value)
+		case "write":
+			write += float64(entry.Value)
+		}
+	}
+	return read, write
+}
+
+// calculateMemoryUsage returns the current usage, limit, and percentage
+func calculateMemoryUsage(stats *libcgroups.MemoryStats) (float64, float64, float64) {
+	// Note that there is also MaxUsage
+	memUsage := float64(stats.Usage.Usage)
+	memLimit := 0.0
+	memPercent := 0.0
+
+	// Calculate total memory of system
+	in := &syscall.Sysinfo_t{}
+	err := syscall.Sysinfo(in)
+	if err == nil {
+		memLimit = float64(in.Totalram) * float64(in.Unit)
+	}
+	if memLimit != 0 {
+		memPercent = memUsage / memLimit * 100.0
+	}
+	return memUsage, memLimit, memPercent
+}
+
+// InstanceStats uses underlying cgroups to get statistics for a named instance
+func InstanceStats(name, instanceUser string, formatJSON bool) error {
+	ii, err := instanceListOrError(instanceUser, name)
+	if err != nil {
+		return err
+	}
+	// Instance stats required 1 instance
+	if len(ii) != 1 {
+		return fmt.Errorf("query returned more than one instance (%d)", len(ii))
+	}
+
+	// Grab our instance to interact with!
+	i := ii[0]
+	if !formatJSON {
+		sylog.Infof("Stats for %s instance of %s (PID=%d)\n", i.Name, i.Image, i.Pid)
+	}
+
+	// Cut out early if we do not have cgroups
+	if !i.Cgroup {
+		url := "https://sylabs.io/guides/latest/user-guide/cgroups.html"
```

Comment:
Could you change this to point people at the "user guide instances section" or similar.

A direct URL is nice, but will become misleading at some point. Many people trail quite a long way behind the latest version of SingularityCE. When we make rootless cgroups for instances the default, and update the docs, etc.  then the information there would be misleading for older versions.

We would likely document `instance stats` in the instances section of the guide, including a summary of the cgroups requirement, and links to more detail. It's called `Running Services` in the contents, at present - but needs to be changed so it's more obviously the instances section.... `Running Services (Instances)` or similar.